worlds/kh2/Rules.py

```diff
@@ -355,6 +355,16 @@ def __init__(self, world: KH2World) -> None:
             RegionName.Master: lambda state: self.multi_form_region_access(),
             RegionName.Final:  lambda state: self.final_form_region_access(state)
         }
+        # Accessing Final requires being able to reach one of the locations in final_leveling_access, but reaching a
+        # location requires being able to reach the region the location is in, so an indirect condition is required.
+        # The access rules of each of the locations in final_leveling_access do not check for being able to reach other
+        # locations or other regions, so it is only the parent region of each location that needs to be added as an
+        # indirect condition.
+        self.form_region_indirect_condition_regions = {
+            RegionName.Final: {
+                self.multiworld.get_location(location, self.player).parent_region for location in final_leveling_access
```

Comment:
Might as well use the helper
```suggestion
                self.get_location(location).parent_region for location in final_leveling_access
```

Reply:
Heh, that's what I get for copy-pasting.

Reply:
Oh, I just realized what `rules` is inheriting from 🤦 `self.world.get_location` would be what's needed